doxylicious/README.md

```diff
@@ -0,0 +1,229 @@
+satisfies: installing_the_thing
+# ... title, procedure, etc
+---
+id: installing_the_thing_using_dnf
+satisfies: installing_the_thing
+# ... title, procedure, etc
+```
+
+Aside: `notional` and `satisfies` are conceptually the same as `interface` and `implements` in the Java programming language. Procedures already use the term `abstract`, in the sense of a summary, so different terminology is used to avoid ambiguity.
+
+When testing, a `notional` procedure can be replaced by any of the procedures which satisfy it. 
+By using a `notional` prerequisite procedure it's possible to execute the same procedure multiple times, but with the prerequisite satisfied in different ways.
+
+```yaml
+id: using_the_thing
+prerequisites:
+  - ref: installing_the_thing
+# ...
+```
+
+In this example, `using_the_thing` can be tested twice: once with the `installing_the_thing` prerequisite satisfied by `installing_the_thing_from_the_website`, and again with it satisfied by `installing_the_thing_using_dnf`. You can choose to execute tests covering all the available satisfying procedures, or to select a subset.
+
+## Testing
+
+### Configure the dependency
+
+Add the dependency in `test` scope, in the `project/dependencies` of your `pom.xml`:
+```xml
+<dependency>
+    <groupId>io.kroxylicious</groupId>
+    <artifactId>doxylicious-junit-provider</artifactId>
+    <version>${project.version}</version>
+    <scope>test</scope>
+</dependency>
+```
+
+### Write your procedure YAMLs
+
+Put them in the `src/test/proc` directory.
+
+### Write your test classes
+
+The extension works with `@TestTemplate`-annotated methods and provides
+an argument for test method parameters with type `Procedure`.
+The extension knows about all the procedures defined in `src/main/proc` and `src/test/proc`.
+Those procedures will be validated by the extension to detect things like procedures with cyclic definitions.
+The method must also be annotated with `@TestProcedure` so the extension knows  which procedure is being tested.
+The extension will arrange for the execution of the test procedure's transitive `prerequisites` before entry to the test method, and also for the execution of `tearDown` steps of the test procedure and its transitive prerequisites. 
+
+The test method will be invoked multiple times, once for each of the possible ways the test procedure and its prerequisites can be satisfied. 
+
+The `@TestProcedure` annotation can also be used to skip execution of some of the prerequisites of the test procedure (for example if the test class itself is honoring a prerequisite via a `@BeforeEach`-annotated method).
+
+The `Procedure` object can be used to execute the `procedure` and to assert that its `verification` is successful.
+
+```java
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.kroxylicious.doxylicious.junit5.*;
+
+class ExampleTest {
+
+    @TestTemplate
+    @ExtendWith(ProcedureTestTemplateExtension.class)
```

Comment:
I think we can make this implicit when adding `@TestProcedure` by adding it to `@TestProcedure`.

Reply:
Oooh, maybe we can. I hadn't noticed that you can annotate the annotation. Thanks!